File: TheDecoder.cpp
#include <iostream>
#include <string>

int main() {
    std::string encodedLine;
    while (std::getline(std::cin, encodedLine)) {
        std::string decodedLine = "";
        for (int i = 0; i < encodedLine.length(); i++) {
            int x = encodedLine[i] - 7;
            decodedLine += (char) x;
        }
        std::cout << decodedLine << std::endl;
    }
    return 0;
}

File: TheDecoder.java
import java.util.Scanner;

public class TheDecoder {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNextLine()) {
            String encodedLine = scanner.nextLine();
            String decodedLine = "";
            for (int i = 0; i < encodedLine.length(); i++) {
                int x = encodedLine.charAt(i) - 7;
                char ch = (char) (x);
                decodedLine = decodedLine + String.valueOf(ch);
            }
            System.out.println(decodedLine);
        }
        scanner.close();
    }
}
